commons/src/main/scala/com/mesosphere/usi/async/ThreadPoolContext.scala

```diff
@@ -0,0 +1,36 @@
+package com.mesosphere.usi.async
+
+import java.util.concurrent.{ExecutorService, Executors, ThreadFactory}
+// TODO: Should I import Guava?
+import com.google.common.util.concurrent.ThreadFactoryBuilder
```

Comment:
@timcharper, is there another elegant way to create a thread poll factory?

Reply:
Not sure, I haven't done this much